central/sensor/service/connection/connection_impl.go

```diff
@@ -155,8 +156,13 @@ func (c *sensorConnection) runRecv(ctx context.Context, grpcServer central.Senso
 
 func (c *sensorConnection) handleMessages(ctx context.Context, queue *dedupingQueue) {
 	for msg := queue.pullBlocking(&c.stopSig); msg != nil; msg = queue.pullBlocking(&c.stopSig) {
-		if err := c.handleMessage(ctx, msg); err != nil {
-			log.Errorf("Error handling sensor message: %v", err)
+		err := safe.Run(func() {
```

Comment:
why Run vs RunE?

Reply:
RunE does not allow me to distinguish between the types of errors and there are many times we have errors handling sensor messages